dags/common/sftp_service.py

```diff
@@ -55,9 +57,15 @@ def __exit__(self, exc_type, exc_value, tb):
             return False
         return True
 
+    def store_files_name(self, fname):
+        self.file_names.append(re.sub(self.dir + "/", "", fname))
+
     def list_files(self):
         try:
-            return self.connection.listdir(self.dir)
+            self.connection.walktree(
+                self.dir, self.store_files_name, lambda x: x, lambda x: x, recurse=True
```

Comment:
why we need the `lambda`s?

Reply:
yes, because all 3 callbacks are required arguments